packages/babel-helper-create-class-features-plugin/src/decorators.ts

```diff
@@ -754,12 +842,14 @@ function createPrivateBrandCheckClosure(brandName: t.PrivateName) {
 // Check if the expression does not reference function-specific
 // context or the given identifier name.
 // `true` means "maybe" and `false` means "no".
-function usesFunctionContext(expression: t.Node) {
+function usesFunctionContextOrYieldAwait(expression: t.Node) {
   try {
     t.traverseFast(expression, node => {
       if (
         t.isThisExpression(node) ||
         t.isSuper(node) ||
+        t.isYieldExpression(node) ||
+        t.isAwaitExpression(node) ||
```

Comment:
This function determines whether the class / element decorator evaluations should be moved into the `applyDecs` static block. If the decorator expression contains `await` or `yield`, we should not move it since the static block can not provide correct function context.